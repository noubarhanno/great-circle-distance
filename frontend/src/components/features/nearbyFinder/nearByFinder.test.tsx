import React from "react";
import { render } from "@testing-library/react";
import NearbyFinderForm from "./form";

describe("Nearby finder form test suite", () => {
  it("should match snopshot", () => {
    const mochFunction = jest.fn();
    const submitMochFunction = jest.fn();
    const { container } = render(
      <NearbyFinderForm
        lat="0"
        lon="0"
        range="0"
        onChange={mochFunction}
        onSubmit={submitMochFunction}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should trigger the submit and show correct label on the button", () => {
    const mochFunction = jest.fn();
    const submitMochFunction = jest.fn();
    const { getByTestId } = render(
      <NearbyFinderForm
        lat="0"
        lon="0"
        range="0"
        onChange={mochFunction}
        onSubmit={submitMochFunction}
      />
    );
    const submitButton = getByTestId("submit-button");
    submitButton.click();
    expect(submitMochFunction).toHaveBeenCalledTimes(1);
    expect(submitButton).toHaveTextContent("Get nearby partners");
  });

  // add more tests here
});
