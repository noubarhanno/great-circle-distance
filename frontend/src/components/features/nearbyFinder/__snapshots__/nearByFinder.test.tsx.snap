// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nearby finder form test suite should match snopshot 1`] = `
<div>
  <div
    style="opacity: 0; transform: translateY(-50px) translateZ(0);"
  >
    <form
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-14my7qk-MuiPaper-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-9dsl64-MuiTypography-root"
      >
        Enter the meeting location
      </h6>
      <div
        class="MuiBox-root css-186wd1t"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            Meeting Latitude
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              name="lat"
              type="text"
              value="0"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Meeting Latitude
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            Meeting Longtude
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              name="lon"
              type="text"
              value="0"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Meeting Longtude
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r2:"
          id=":r2:-label"
        >
          Range
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r2:"
            name="range"
            type="text"
            value="0"
          />
          KM
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Range
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-nheg8e-MuiButtonBase-root-MuiButton-root"
        data-testid="submit-button"
        tabindex="0"
        type="submit"
      >
        Get nearby partners
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
